accepted/2021/architecture-targeting.md

```diff
@@ -0,0 +1,498 @@
+* The presence of the apphost for a portable app might be useful for development, but it is confusing in a deployed environment if the development and production RIDs do not match. If they do match, then the app isn't as optimized as it could be.
+* Portable apps will be bigger if the app depends on packages with architecture-specific dependencies, since the app will (likely) contain the native assets for multiple architectures (that's one of three key aspects of what makes it portable). The only reason to build portable apps is to accommodate multi-architecture packages.
+* Portable apps and RID-specific builds [do not place native assets in the same location](https://github.com/dotnet/sdk/issues/11754). That should be configurable and is a [confusing behavior](https://github.com/dotnet/sdk/issues/3903).
+* It is possible to make an app portable for one operating system, such as for Windows x86, x64, and Arm64, but that isn't enabled.
+* Portable apps -- for the apphost -- don't support the universal binaries feature of Windows or macOS, which reduces their utility, particularly for client apps.
+* Some users want to target a specific architecture and/or operating system. The SDK produces a self-contained app by default when you do this. Many users (likely most) do not want that.
+* For example, we see much higher pulls of our container images for framework-dependent apps as opposed to self-contained ones, which tells us that the framework-dependent modality is very popular.
+* There are important scenarios for targeting a specific architecture with framework-dependent apps. It's a logical leap for the SDK to switch to the self-contained deployment model as a consequence of targeting a specific RID since RID targeting for framework-dependent apps is a legitimate and important scenario.
+* There are performance benefits for framework-dependent apps being RID-specific.
+
+### Ugly
+
+* `dotnet build` and `dotnet publish` do not have the same capabilities (in terms of CLI arguments) for architecture targeting. Both verbs enable specifying a RID, while only `dotnet publish` enables changing the deployment model, between self-contained (the default when `-r` is used) and framework-dependent. This isn't technically true, because you can rely on MSBuild properties for this purpose. It is, however, effectively true.
+
+## Desired build types
+
+Today, we talk about self-contained and framework-dependent as the two [deployment types](https://docs.microsoft.com/dotnet/core/deploying/). This model is misleading for two different reasons: the set isn't complete, and it's oriented exclusively on deployment.
+
+The following are the full set of "build types" (I'm changing the term) that can be used to greater or lesser degrees across development, testing, and production scenarios.
+
+* Portable framework-dependent app
+* RID-specific framework-dependent app
+* RID-specific self-contained app
+
+Note: The first build type is the default today.
```

Comment:
I don't think this is true (e.g. Client Blazor) and in .NET7 e.g. MAUI

Reply:
Totally true. My thinking is that these app types already have bespoke build experiences so many of these default experiences don't apply. Fair?